capa/features/extractors/dnfile/insn.py

```diff
@@ -89,36 +101,110 @@ def extract_insn_api_features(fh: FunctionHandle, bh, ih: InsnHandle) -> Iterato
         yield API(str(callee)), ih.address
 
 
+def extract_insn_property_features(fh: FunctionHandle, bh, ih: InsnHandle) -> Iterator[Tuple[Feature, Address]]:
+    """parse instruction property features"""
+    insn: Instruction = ih.inner
+
+    if insn.opcode in (OpCodes.Call, OpCodes.Callvirt, OpCodes.Jmp, OpCodes.Calli):
+        token: Token = Token(insn.operand.value)
+        if token.table == 6:
+            property: Union[DnProperty, None] = get_dotnet_property(fh.ctx["pe"], token)
+            if property is None:
+                return
+            yield Property(str(property)), ih.address
+
+        elif token.table == 10:
+            row: Any = resolve_dotnet_token(fh.ctx["pe"], token)
+            if row is None:
+                return
+            if row.Name.startswith("get_"):
+                name = row.Name.replace("get_", "")
+            elif row.Name.startswith("set_"):
+                name = row.Name.replace("set_", "")
+            else:
+                return
+            if not isinstance(row.Class.row, (dnfile.mdtable.TypeRefRow, dnfile.mdtable.TypeDefRow)):
+                return
+            yield Property(
+                DnProperty.format_name(row.Class.row.TypeNamespace, row.Class.row.TypeName, name)
+            ), ih.address
+
+    if insn.opcode in (OpCodes.Ldfld, OpCodes.Ldflda, OpCodes.Ldsfld, OpCodes.Ldsflda, OpCodes.Stfld, OpCodes.Stsfld):
+        field_token: Token = Token(insn.operand.value)
+        if field_token.table == 4:
+            for field in get_dotnet_fields(fh.ctx["pe"]):
+                if field_token.value == field.token:
+                    yield Property(str(field)), ih.address
+
+    return
+
+
 def extract_insn_class_features(fh: FunctionHandle, bh, ih: InsnHandle) -> Iterator[Tuple[Class, Address]]:
     """parse instruction class features"""
-    if ih.inner.opcode not in (OpCodes.Call, OpCodes.Callvirt, OpCodes.Jmp, OpCodes.Calli):
+    if ih.inner.opcode not in (
+        OpCodes.Call,
+        OpCodes.Callvirt,
+        OpCodes.Jmp,
+        OpCodes.Calli,
+        OpCodes.Ldfld,
+        OpCodes.Ldflda,
+        OpCodes.Ldsfld,
+        OpCodes.Ldsflda,
+        OpCodes.Stfld,
+        OpCodes.Stsfld,
+    ):
         return
 
     row: Any = resolve_dotnet_token(fh.ctx["pe"], Token(ih.inner.operand.value))
+    if isinstance(row, dnfile.mdtable.MemberRefRow):
+        if isinstance(row.Class.row, (dnfile.mdtable.TypeRefRow, dnfile.mdtable.TypeDefRow)):
+            yield Class(DnClass.format_name(row.Class.row.TypeNamespace, row.Class.row.TypeName)), ih.address
 
-    if not isinstance(row, dnfile.mdtable.MemberRefRow):
-        return
-    if not isinstance(row.Class.row, (dnfile.mdtable.TypeRefRow, dnfile.mdtable.TypeDefRow)):
-        return
+    elif isinstance(row, dnfile.mdtable.MethodDefRow):
+        for method in get_dotnet_managed_methods(fh.ctx["pe"]):
```

Comment:
Nice catch - can we use the cached methods here https://github.com/mandiant/capa/blob/2f7aa14f61ee83ccba41b97ff0cbfd15de21b25b/capa/features/extractors/dnfile/insn.py#L52-L57?

Let's also emit `class` and `namespace` features at the file scope https://github.com/mandiant/capa/blob/2f7aa14f61ee83ccba41b97ff0cbfd15de21b25b/capa/features/extractors/dotnetfile.py#L62-L90

Reply:
Noted. https://github.com/mandiant/capa/pull/1122/commits/bf1713d7f47e8a883fed4832065ad0f61d702785 caches fields/properties.